### PR TITLE
Remove unnecessary fixtures

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Some unit test clean up

## 🔖 Changes

- Removed unnecessary test fixtures
- Add more tests for the invalid scalar input types

### CODE REVIEW

**Review by JesusValera — APPROVED** [review]
,🚀🚀🌚

File: tests/Unit/ArrayOfTest.php [hunk 1]
```diff
@@ -33,49 +29,47 @@ public function testKeysArePreserved(): void
             'key2' => 'value2',
         ];
 
-        $test = (array) new ValidScalarArrayOf($input);
+        $test = (array)new ArrayOfString($input);
```
Comment (Tito-Kati) [inline tests/Unit/ArrayOfTest.php:32]: Is this intentional? I guess they are habits, in my case I like more that casts are separated by a space. But I don't care that this stays that way. Could this be configured in the PHP CS Fixer?

File: tests/Unit/ArrayOfTest.php [hunk 2]
```diff
@@ -33,49 +29,47 @@ public function testKeysArePreserved(): void
...
-        new ValidScalarArrayOf([1]);
-    }
-
-    public function testInvalidClassInputType(): void
-    {
-        $this->expectException(InvalidInstantiationType::class);
-        new ValidClassArrayOf([new stdClass()]);
-    }
-
     public function testCanUseAsArray(): void
```
Comment (Tito-Kati) [inline tests/Unit/ArrayOfTest.php:70]: This test for some reason does not convince me, I know it was already like that and I am not saying to fix it now, I just wanted to put it on the record.

**Review by Tito-Kati — APPROVED** [review]
For me it is perfect although I have noted a couple of things that I leave freedom if we do something with them or not. I approve to expedite the process.

File: tests/Unit/ArrayOfTest.php (hunk 1 above)
Comment (Chemaclass) [reply tests/Unit/ArrayOfTest.php:32]: Yeah... I suppose this is part of my habit. I will look for a PHP CS Fixer rule for this one. Thanks for saying it loud :)

File: tests/Unit/ArrayOfTest.php (hunk 2 above)
Comment (Chemaclass) [reply tests/Unit/ArrayOfTest.php:70]: Feel free to remove/update it in a future PR ;)

**Pull request merged**